[docs] eas build: add page about internal distribution (#11124)
expo/expo: docs/pages/build/internal-distribution.md

-->

<!--
(@dsokal) this is not implemented yet

#### Setting up enterprise provisioning

If you do have an Apple enterprise account, this makes internal distribution much easier for users who want to install your app for the first time. Once they install the profile to their device they can access the app right away. One limitation of using an enterprise provisioning profile is that you will need to have a distinct bundle identifier from the one that you use to publish your app to the App Store. (@brentvatne: after this is a bad idea intentionally, we should probably have a config option, i'm just putting it there so we have something for now) We recommend setting your bundle identifier for internal distribution and committing that change on another branch. After that, whenever you want to create a preview branch you can check out that branch and rebase against the branch you'd like to create a build for.
-->

#### Setting up adhoc provisioning

Apps signed with an adhoc provisioning profile can be installed by any iOS device whose unique identifier (UDID) is registered with the provisioning profile.

> Setting up adhoc provisioning consists of two steps. In the first step, you'll register all devices you want to run your app on. Simply run the following command and you'll see the URL you have to open on all those devices (the command will also print a QR code for convenience).

Reviewer: ```suggestion
Setting up adhoc provisioning consists of two steps. In the first step, you'll register all devices you want to run your app on. Run the following command to generate a URL (and QR code, for convenience) that you can open on your devices to register them to be eligible for internal distribution.
```

<TerminalBlock cmd={['# Register Apple Devices for internal distribution', 'eas device:create']} />

> When all your devices are registered the next step is just running a build. The build command will guide you through the credentials (provisioning profile) generation process.

Reviewer: ```suggestion
When all of your devices are registered, the next step is to run a build. The build command will guide you through the credentials (provisioning profile) generation process.
```

#### Advanced flow: Using credentials.json

**Using credentials.json?** If you are using local credentials, make sure you point your `credentials.json` to an ad-hoc <!-- or enterprise -->provisioning profile that you generate through the Apple Developer portal. Beware that EAS CLI does only a limited validation of your local credentials. Use at your own risk!

### Run a build with the internal build profile

Now that we have set up our build profile and app signing, running a build for internal distribution is just like any other build.

<TerminalBlock cmd={['# Create iOS and Android builds for internal distribution', 'eas build:create --profile preview --platform all']} />

If you choose to let Expo manage your adhoc provisioning profile, we will show you the list of all devices registered so far. You will then be prompted to confirm you want to proceed with those devices.

> When the build completes, you will be given a URL that you can share with your team (provided that they have an Expo account and are part of your organization <!-- (@dsokal) we'll change this requirement so everyone who knows the URL can install the app -->) to download and install the app.

Reviewer: ```suggestion
When the build completes, you will be given a URL that you can share with your team (provided that they have an Expo account and are part of your organization) <!-- (@dsokal) we'll change this requirement so everyone who knows the URL can install the app --> to download and install the app.
```

Reviewer: what's missing to make this possible?

Author: Answered on Slack.
